homeassistant/components/vivotek/camera.py

```diff
@@ -0,0 +1,174 @@
+        vol.Optional(CONF_AUTHENTICATION, default=HTTP_BASIC_AUTHENTICATION): vol.In(
+            [HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]
+        ),
+        vol.Optional(CONF_LIMIT_REFETCH_TO_URL_CHANGE, default=False): cv.boolean,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_CONTENT_TYPE, default=DEFAULT_CONTENT_TYPE): cv.string,
+        vol.Optional(CONF_FRAMERATE, default=2): cv.positive_int,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+    }
+)
+
+STATE_DETECTING_MOTION = "detecting motion"
+STATE_IDLE = "idle"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up a generic IP Camera."""
+    async_add_entities([VivotekCamera(hass, config)])
+
+
+class VivotekCamera(Camera):
+    """A Vivotek IP camera."""
+
+    def __init__(self, hass, device_info):
+        """Initialize a generic camera."""
+        from libpyvivotek import VivotekCamera
+
+        super().__init__()
+
+        self.hass = hass
+
+        self._name = device_info.get(CONF_NAME)
+        self._limit_refetch = device_info[CONF_LIMIT_REFETCH_TO_URL_CHANGE]
+        self._frame_interval = 1 / device_info[CONF_FRAMERATE]
+        self.content_type = device_info[CONF_CONTENT_TYPE]
+        self.verify_ssl = device_info[CONF_VERIFY_SSL]
+        self._event_i0_status = None
+        self._event_0_key = DEFAULT_EVENT_0_KEY
+
+        username = device_info.get(CONF_USERNAME)
+        password = device_info.get(CONF_PASSWORD)
+
+        if device_info[CONF_STREAM_SOURCE]:
+            self._stream_source = (
+                "rtsp://%s:%s@%s:554/live.sdp",
+                username,
+                password,
+                device_info[CONF_STREAM_SOURCE],
+            )
+        else:
+            self._stream_source = None
+
+        self._brand = "Vivotek"
+
+        self._supported_features = SUPPORT_STREAM if self._stream_source else 0
+
+        self._last_url = None
+        self._last_image = None
+
+        self._cam = VivotekCamera(
+            host=device_info.get(CONF_IP_ADDRESS),
+            port=443,
+            verify_ssl=device_info[CONF_VERIFY_SSL],
+            usr=username,
+            pwd=password,
+        )
+
+        self._motion_detection_enabled = self.event_enabled(self._event_0_key)
+
+    @property
+    def supported_features(self):
+        """Return supported features for this camera."""
+        return self._supported_features
+
+    @property
+    def frame_interval(self):
+        """Return the interval between frames of the mjpeg stream."""
+        return self._frame_interval
+
+    def event_enabled(self, event_key):
+        """Return true if event for the provided key is enabled."""
+        response = self._cam.get_param(event_key)
+        return int(response.replace("'", "")) == 1
```

Comment:
The library should parse the response from the camera. We shouldn't have to do that here in home assistant.

There should be separate methods for different features and they should return values that can be used without parsing, eg `True` or `False`.